saleor/payment/gateways/braintree/__init__.py

```
@@ -0,0 +1,265 @@
+            kind=TransactionKind.CAPTURE,
+            token=auth_transaction.token,
+            amount=amount)
+
+    gateway_response = extract_gateway_response(result)
+    error = get_error_for_client(gateway_response['errors'])
+
+    txn = create_transaction(
+        payment=payment,
+        kind=TransactionKind.CAPTURE,
+        amount=gateway_response.pop('amount', amount),
+        currency=gateway_response.pop('currency', payment.currency),
+        token=getattr(result.transaction, 'id', ''),
+        is_success=result.is_success,
+        gateway_response=gateway_response)
+    return txn, error
+
+
+def void(
+        payment: Payment,
+        **connection_params: Dict) -> Tuple[Transaction, str]:
+    gateway = get_braintree_gateway(**connection_params)
+    auth_transaction = payment.transactions.filter(
+        kind=TransactionKind.AUTH, is_success=True).first()
+    try:
+        result = gateway.transaction.void(
+            transaction_id=auth_transaction.token)
+    except braintree_sdk.exceptions.NotFoundError:
+        return transaction_and_incorrect_token_error(
+            payment, kind=TransactionKind.VOID, token=auth_transaction.token)
+
+    gateway_response = extract_gateway_response(result)
+    error = get_error_for_client(gateway_response['errors'])
+    txn = create_transaction(
+        payment=payment,
+        kind=TransactionKind.VOID,
+        amount=gateway_response.pop('amount', payment.total),
+        currency=gateway_response.pop('currency', payment.currency),
+        gateway_response=gateway_response,
+        token=getattr(result.transaction, 'id', ''),
+        is_success=result.is_success)
+    return txn, error
+
+
+def refund(
+        payment: Payment,
+        amount: Decimal,
+        **connection_params: Dict) -> Tuple[Transaction, str]:
+    gateway = get_braintree_gateway(**connection_params)
+    capture_txn = payment.transactions.filter(
+        kind=TransactionKind.CAPTURE, is_success=True).first()
```

Should we handle the case where `capture_txn` is `None`? Maybe return an error?